scene/2d/cpu_particles_2d.cpp

```diff
@@ -965,7 +966,10 @@ void CPUParticles2D::_update_render_thread() {
 void CPUParticles2D::_notification(int p_what) {
 
 	if (p_what == NOTIFICATION_ENTER_TREE) {
-		_set_redraw(true);
+		if (emitting)
+			set_process_internal(true);
+		else
+			set_process_internal(false);
```

Comment:
this could be simply `set_process_internal(emitting);`

Reply:
Ah it really is, will changed the commit :)